src/features/defi/providers/fox-farming/components/FoxFarmingManager/Deposit/components/Status.tsx

```diff
@@ -94,25 +102,100 @@ export const Status: React.FC<StatusProps> = ({ accountId }) => {
     if (!(state?.txid && accountId && accountAddress)) return ''
     return serializeTxIndex(accountId, state.txid, accountAddress)
   }, [state?.txid, accountAddress, accountId])
+
   const confirmedTransaction = useAppSelector(gs => selectTxById(gs, serializedTxIndex))
 
+  const { statusIcon, status, statusText, statusBg, statusBody } = useMemo(() => {
+    if (maybeSafeTx?.isQueuedSafeTx)
+      return {
+        statusIcon: null,
+        statusText: StatusTextEnum.pending,
+        status: TxStatus.Pending,
+        statusBody: translate('common.safeProposalQueued', {
+          currentConfirmations: maybeSafeTx?.transaction?.confirmations?.length,
```

Comment:
There seems to be a lot of duplicate code surrounding the `Status` components, perhaps we can consolidate into helpers? Not for this PR of course, it's been open too long already!

Reply:
Added to my TODOs, happy to follow-up with that! 🙏🏽